python/tank/bootstrap/configuration.py

```diff
@@ -137,15 +159,14 @@ def _set_authenticated_user(self, user, serialized_user):
         If the new core API can't deserialize the user, the error will be logged and passed in
         user will be used instead.
 
-        When this method is done executing,
-
-        1. If the project has a script user, that script will be the authenticated user.
-        2. If the project didn't have a script user, whoever was used to bootstrap
-           will be the authenticated user.
-
         :param user: User that was used for bootstrapping.
         :param serialized_user: Serialized version of the user.
+
+        :returns: If authentication is supported, a :class:`ShotgunUser` will be returned.
         """
+        # perform a local import here to make sure we are getting
+        # the newly swapped in core code
+
```

Comment:
Hmm, this goes back to previous discussions and again apologies if this is crazy: 

The 'clean' and default way to handle this should be via serialization right? So why don't we start by doing that? And in case that fails, we fall back on custom logic and special tricks to handle edge cases:

```
def _set_authenticated_user(self, bootstrap_user, bootstrap_user_login, serialized_user):
   try:
      user = deserialize_user(serialized_user)
   except:
      # for some reason couldn't do the right thing
   # fall back on special tricks
```

Reply:
Because the project's configuration might be different than the global settings.
- Different proxy
- Different api-endpoint
- Using a script user

It's all in that wall of comments.

Reply:
No, it's not actually. I'll beef up the comments.